worker/include/RTC/RtpPacket.hpp

```diff
@@ -24,6 +22,9 @@ namespace RTC
 	class RtpPacket
 	{
 	public:
+		using RtpPacketBuffer = std::array<uint8_t, MtuSize + 100>;
+		using SharedPtr       = std::shared_ptr<RtpPacket>;
```

Comment:
`std::shared_ptr` does an extra memory allocation for the reference counter area. Has that been taken into account?
I implemented something like shared_ptr myself because I seemed to care about the one time extra global allocation cost as well.

Reply:
We are not allocating memory for the shared pointer itself, just for what it contains, so we are safe.

Reply:
Reference count variable still needs to be allocated somewhere (older version kept it in the original data structure since code is single-threaded and packet is already heap-allocated), so I think @satoren is right, but since it happens once for all shared instances it is probably worth the tiny performance cost to save on maintenance later.

Reply:
I created [pull request](https://github.com/nazar-pc/mediasoup/pull/8) which removes allocation for ```std::shared_ptr```